File: app/src/main/java/com/wzq/sample/ui/room_db/RoomSampleViewModel.java
package com.wzq.sample.ui.room_db;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;

import com.wzq.sample.base.BaseViewModel;

import java.util.List;


/**
 * Create Date：2020/01/01
 *  实现Room数据的基本操作
 *  王志强
 */
public class RoomSampleViewModel extends BaseViewModel {

    private WordRepository wordRepository;

    public RoomSampleViewModel(@NonNull Application application) {
        super(application);
        wordRepository = new WordRepository(application);
    }

    LiveData<List<Word>> getAllWordsLive() {

        return wordRepository.getAllWordsLive();
    }

    void insertWords(Word... words) {
        wordRepository.insertWords(words);
    }

    void updateWords(Word... words) {
        wordRepository.updateWords(words);
    }

    void deleteWords(Word... words) {
        wordRepository.deleteWords(words);
    }

    void deleteAllWords() {
        wordRepository.deleteAllWords();
    }

}

File: app/src/main/java/com/wzq/sample/ui/room_db/RoomSampleViewModel.kt
package com.wzq.sample.ui.room_db

import android.app.Application
import androidx.lifecycle.LiveData
import com.wzq.sample.base.BaseViewModel

/**
 * Create Date：2020/01/01
 * 实现Room数据的基本操作
 * 王志强
 */
class RoomSampleViewModel(application: Application) : BaseViewModel(application) {
    private val wordRepository: WordRepository = WordRepository(application)
    val allWordsLive: LiveData<List<Word>>
        get() = wordRepository.listLiveData

    fun insertWords(vararg words: Word?) {
        wordRepository.insertWords(*words)
    }

    fun updateWords(vararg words: Word?) {
        wordRepository.updateWords(*words)
    }

    fun deleteWords(vararg words: Word?) {
        wordRepository.deleteWords(*words)
    }

    fun deleteAllWords() {
        wordRepository.deleteAllWords()
    }

}